eks-public-cluster.tf

```diff
@@ -117,3 +117,9 @@ data "aws_eks_cluster" "public-cluster" {
 data "aws_eks_cluster_auth" "public-cluster" {
   name = module.eks-public.cluster_id
 }
+
+provider "kubernetes" {
```

Comment:
- Should be in the `providers.tf`
- The file `.terraform.lock.hcl` need to be updated to pin the current version of this new provider (you can check the log of the Terraform plan in the CI check (`Finding hashicorp/kubernetes versions matching ">= 2.10.0"`)